source/extensions/common/dynamic_forward_proxy/dns_cache.h

```diff
@@ -177,6 +177,14 @@ class DnsCache {
    */
   virtual absl::flat_hash_map<std::string, DnsHostInfoSharedPtr> hosts() PURE;
 
+  /**
+   * Retrieve the DNS host info of a given host currently stored in the cache.
+   * @param host_name supplies the host name.
+   * @return the DNS host info associated with the given host name if such a host is currently
```

Comment:
minor nit
consider
if such a host is currently stored in the cache ->
if the host's address is cached
since we don't return the DNS host if the host is in cache but the address is unresolved.

Reply:
Done in commit 7b426d8c5